client/dashboard/components/opt-in-welcome/index.tsx

```diff
@@ -0,0 +1,33 @@
+import { userPreferenceQuery, userPreferenceMutation } from '@automattic/api-queries';
+import { useMutation, useSuspenseQuery } from '@tanstack/react-query';
+import { Link } from '@tanstack/react-router';
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import Notice from '../../components/notice';
+
+export function OptInWelcome() {
+	const { data: isDismissedPersisted } = useSuspenseQuery(
+		userPreferenceQuery( 'hosting-dashboard-welcome-notice-dismissed' )
+	);
+	const { mutate: dismiss, isPending: isDismissing } = useMutation(
+		userPreferenceMutation( 'hosting-dashboard-welcome-notice-dismissed' )
+	);
+
+	// Optimistically hide the banner assuming the preference will get saved.
+	if ( isDismissing || isDismissedPersisted ) {
+		return null;
+	}
+
+	return (
+		<Notice onClose={ () => dismiss( new Date().toISOString() ) } variant="promote">
+			{ createInterpolateElement(
+				__(
+					'Welcome to your new hosting dashboard. To switch back to the previous version, go to <preferencesLink>Preferences</preferencesLink>.'
+				),
+				{
+					preferencesLink: <Link to="/me/preferences" />,
```

Comment:
I tried linking to something like to `/me/preferences#the-correct-location-on-page`. It wasn't great though. The floating header gets in the way so that the preferences card didn't look like it was at the top of the page.

The preferences card is at the top of the page for now, so it's fine for now.